.github/workflows/java_client_compatibility.yaml

```diff
@@ -81,13 +81,29 @@ jobs:
         run: |
           echo "HZ_LICENSEKEY=${{ '>' == steps.version.outputs.comparison-result && secrets.HAZELCAST_ENTERPRISE_KEY || secrets.HAZELCAST_ENTERPRISE_KEY_V5 }}" >> $GITHUB_ENV
 
+      - name: Build modules
+        shell: bash -l {0}
+        run: |
+          chmod +x mvnw
+          ./mvnw -B -V -e clean install -DskipTests
+        working-directory: client
+
       - name: Run non-enterprise tests
+        if: ${{ matrix.kind == 'os' }}
```

Comment:
you should still run the OS related tests(e.g. IMap tests) against the EE server.

Reply:
I added them as a separate job